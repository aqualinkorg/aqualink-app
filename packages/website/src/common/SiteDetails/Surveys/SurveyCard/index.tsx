import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Paper,
  Grid,
  CardMedia,
  Typography,
  Button,
} from "@material-ui/core";
import { Link } from "react-router-dom";

import { formatNumber } from "../../../../helpers/numberUtils";
import DeleteButton from "../DeleteButton";
import { SurveyListItem } from "../../../../store/Survey/types";
import incomingStyles from "../styles";
import CustomLink from "../../../Link";

const SurveyCard = ({
  pointId,
  pointName,
  isAdmin,
  siteId,
  survey,
  classes,
}: SurveyCardProps) => {
  const isShowingFeatured = pointId === -1;

  return (
    <Paper elevation={0} className={classes.surveyCard}>
      <Grid style={{ height: "100%" }} container justify="space-between">
        <Grid className={classes.cardImageWrapper} item xs={12} md={5}>
          <Link to={`/sites/${siteId}/survey_details/${survey.id}`}>
            <CardMedia
              className={classes.cardImage}
              image={
                isShowingFeatured
                  ? survey.featuredSurveyMedia?.url
                  : survey.surveyPointImage?.[pointId]?.[0]
              }
            />
          </Link>
        </Grid>
        <Grid className={classes.infoWrapper} container item xs={12} md={7}>
          <Grid className={classes.info} container item xs={12}>
            {survey.user?.fullName && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  User:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {survey.user.fullName}
                </Typography>
              </Grid>
            )}
            {survey.featuredSurveyMedia?.surveyPoint?.name && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  Survey Point:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin} ${classes.blueText}`}
                  variant="h6"
                  title={survey.featuredSurveyMedia.surveyPoint.name}
                >
                  <CustomLink
                    to={`/sites/${siteId}/points/${
                      isShowingFeatured
                        ? survey.featuredSurveyMedia.surveyPoint.id
                        : pointId
                    }`}
                    isIcon={false}
                    tooltipTitle=""
                    content={
                      pointName === "All"
                        ? survey.featuredSurveyMedia.surveyPoint.name
                        : pointName
                    }
                  />
                </Typography>
              </Grid>
            )}
            {survey.comments && (
              <Grid
                className={classes.commentsWrapper}
                container
                alignItems="flex-start"
                item
                xs={12}
              >
                <Grid style={{ height: "80%" }} item xs={12}>
                  <Typography className={classes.cardFields} variant="h6">
                    Comments:
                  </Typography>
                  <Typography
                    className={`${classes.cardValues} ${classes.comments}`}
                    variant="h6"
                  >
                    {survey.comments}
                  </Typography>
                </Grid>
              </Grid>
            )}
            {survey.temperature && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  Temp:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {`${formatNumber(survey.temperature, 1)} Â°C`}
                </Typography>
              </Grid>
            )}
            <Grid
              container
              alignItems="center"
              justify="space-between"
              item
              xs={12}
            >
              <Grid item xs={10}>
                <Link
                  style={{ color: "inherit", textDecoration: "none" }}
                  to={`/sites/${siteId}/survey_details/${survey.id}`}
                >
                  <Button size="small" variant="outlined" color="primary">
                    VIEW DETAILS
                  </Button>
                </Link>
              </Grid>
              {isAdmin && (
                <Grid container justify="flex-end" item xs={2}>
                  <DeleteButton
                    siteId={siteId}
                    surveyId={survey.id}
                    diveDate={survey.diveDate}
                  />
                </Grid>
              )}
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Paper>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    ...incomingStyles,
    cardImageWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    infoWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    commentsWrapper: {
      maxHeight: "51%",
    },
    comments: {
      height: "100%",
      overflowY: "auto",
    },
    valuesWithMargin: {
      marginLeft: "1rem",
      maxWidth: "60%",
      display: "block",
      overflow: "hidden",
      textOverflow: "ellipsis",
      whiteSpace: "nowrap",
    },
    blueText: {
      color: theme.palette.primary.main,
    },
    info: {
      height: "100%",
      padding: "0.5rem 0.5rem 0.5rem 1rem",
    },
  });

interface SurveyCardIncomingProps {
  pointId: number;
  pointName: string | null;
  isAdmin: boolean;
  siteId: number;
  survey: SurveyListItem;
}

type SurveyCardProps = SurveyCardIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(SurveyCard);
