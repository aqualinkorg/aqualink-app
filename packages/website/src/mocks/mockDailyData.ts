import { DailyData } from "../store/Reefs/types";

const now = new Date();
const minutesAgo = 5;
const dailyDataDate = new Date(
  now.getTime() - minutesAgo * 60000
).toISOString();

export const mockDailyData: DailyData = {
  id: 1,
  date: dailyDataDate,
  minBottomTemperature: 25,
  maxBottomTemperature: 27,
  avgBottomTemperature: 26,
  degreeHeatingDays: 29,
  surfaceTemperature: 36,
  satelliteTemperature: 20,
  minWaveHeight: 1,
  maxWaveHeight: 3,
  avgWaveHeight: 2,
  waveDirection: 136,
  wavePeriod: 15,
  minWindSpeed: 3,
  maxWindSpeed: 5,
  avgWindSpeed: 4,
  windDirection: 96,
  weeklyAlertLevel: 3,
};
