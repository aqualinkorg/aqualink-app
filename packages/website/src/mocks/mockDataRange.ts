import { TimeSeriesDataRange } from 'store/Sites/types';

export const mockDataRange: TimeSeriesDataRange = {
  bottomTemperature: [
    {
      type: 'hobo',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  cholorophyllConcentration: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  odoConcentration: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  ph: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  salinity: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
    {
      type: 'hui',
      depth: null,
      data: [
        {
          minDate: '2022-06-11T00:00:00.000Z',
          maxDate: '2022-06-14T23:45:00.000Z',
        },
      ],
    },
  ],
  turbidity: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
};
