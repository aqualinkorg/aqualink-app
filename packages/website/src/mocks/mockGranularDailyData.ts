import { DailyData } from 'store/Sites/types';

export const mockGranularDailyData: DailyData[] = [
  {
    id: 3,
    date: '2022-06-27T02:59:59.999Z',
    minBottomTemperature: 0,
    maxBottomTemperature: 0,
    avgBottomTemperature: 0,
    degreeHeatingDays: 0,
    topTemperature: 0,
    satelliteTemperature: 24.219252436179815,
    minWaveHeight: 0.17217956015502844,
    maxWaveHeight: 0.4220004534334179,
    avgWaveHeight: 0.28517965469411594,
    waveMeanDirection: 219,
    waveMeanPeriod: 2,
    minWindSpeed: 1.0233644143488965,
    maxWindSpeed: 5.558413659400355,
    avgWindSpeed: 2.9796803902255204,
    windDirection: 218,
    weeklyAlertLevel: 0,
  },
  {
    id: 2,
    date: '2022-06-26T02:59:59.999Z',
    minBottomTemperature: 0,
    maxBottomTemperature: 0,
    avgBottomTemperature: 0,
    degreeHeatingDays: 0,
    topTemperature: 0,
    satelliteTemperature: 24.229605847456323,
    minWaveHeight: 0.06706502244580449,
    maxWaveHeight: 0.40127747690701104,
    avgWaveHeight: 0.20286431769935753,
    waveMeanDirection: 186,
    waveMeanPeriod: 2,
    minWindSpeed: 1.0683246478923132,
    maxWindSpeed: 3.8642400151240097,
    avgWindSpeed: 2.4470019134494967,
    windDirection: 238,
    weeklyAlertLevel: 0,
  },
  {
    id: 1,
    date: '2022-06-25T02:59:59.999Z',
    minBottomTemperature: 0,
    maxBottomTemperature: 0,
    avgBottomTemperature: 0,
    degreeHeatingDays: 0,
    topTemperature: 0,
    satelliteTemperature: 24.00602354214886,
    minWaveHeight: 0.25399888336189147,
    maxWaveHeight: 0.7053866500812253,
    avgWaveHeight: 0.49466218581824045,
    waveMeanDirection: 184,
    waveMeanPeriod: 3,
    minWindSpeed: 2.3608638556613584,
    maxWindSpeed: 6.910529333003823,
    avgWindSpeed: 4.019357707456016,
    windDirection: 125,
    weeklyAlertLevel: 0,
  },
];
