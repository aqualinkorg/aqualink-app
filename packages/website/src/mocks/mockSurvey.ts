import { SurveyListItem } from "../store/Survey/types";

export const mockSurvey: SurveyListItem = {
  comments: "No comments",
  diveDate: "2020-09-10T10:27:00.000Z",
  id: 46,
  temperature: 10,
  weatherConditions: "calm",
  observations: ["anthropogenic"],
  userId: {
    id: 0,
    fullName: "Joe Doe",
  },
  featuredSurveyMedia: {
    comments: null,
    featured: true,
    id: 66,
    observations: "possible-disease",
    type: "image",
    url: "",
  },
};
