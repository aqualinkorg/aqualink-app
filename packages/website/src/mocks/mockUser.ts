import { User } from "../store/User/types";

export const mockUser: User = {
  adminLevel: "default",
  administeredReefs: [],
  email: "mail@mail.com",
  firebaseUid: "RaNdOm StRiNg",
  fullName: "Mock User",
  organization: "Some organization",
  token: "RaNdOm StRiNg",
};
