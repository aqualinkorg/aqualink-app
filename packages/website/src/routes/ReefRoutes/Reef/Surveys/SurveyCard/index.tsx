import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Paper,
  Grid,
  CardMedia,
  Typography,
  Button,
} from "@material-ui/core";
import { Link } from "react-router-dom";

import { formatNumber } from "../../../../../helpers/numberUtils";
import DeleteButton from "../DeleteButton";
import { SurveyListItem } from "../../../../../store/Survey/types";
import incomingStyles from "../styles";

const SurveyCard = ({ isAdmin, reefId, survey, classes }: SurveyCardProps) => {
  return (
    <Paper elevation={0} className={classes.surveyCard}>
      <Grid style={{ height: "100%" }} container justify="space-between">
        <Grid className={classes.cardImageWrapper} item xs={12} md={5}>
          {survey.featuredSurveyMedia && (
            <CardMedia
              className={classes.cardImage}
              image={survey.featuredSurveyMedia.url}
            />
          )}
        </Grid>
        <Grid className={classes.infoWrapper} container item xs={12} md={7}>
          <Grid className={classes.info} container item xs={12}>
            {survey.userId!.fullName && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  User:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {survey.userId!.fullName}
                </Typography>
              </Grid>
            )}
            {survey.comments && (
              <Grid
                className={classes.commentsWrapper}
                container
                alignItems="flex-start"
                item
                xs={12}
              >
                <Grid style={{ height: "80%" }} item xs={12}>
                  <Typography className={classes.cardFields} variant="h6">
                    Comments:
                  </Typography>
                  <Typography
                    className={`${classes.cardValues} ${classes.comments}`}
                    variant="h6"
                  >
                    {survey.comments}
                  </Typography>
                </Grid>
              </Grid>
            )}
            {survey.temperature && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  Temp:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {`${formatNumber(survey.temperature, 1)} Â°C`}
                </Typography>
              </Grid>
            )}
            <Grid
              container
              alignItems="center"
              justify="space-between"
              item
              xs={12}
            >
              <Grid item xs={10}>
                <Link
                  style={{ color: "inherit", textDecoration: "none" }}
                  to={`/reefs/${reefId}/survey_details/${survey.id}`}
                >
                  <Button size="small" variant="outlined" color="primary">
                    VIEW DETAILS
                  </Button>
                </Link>
              </Grid>
              {isAdmin && (
                <Grid container justify="flex-end" item xs={2}>
                  <DeleteButton
                    reefId={reefId}
                    surveyId={survey.id}
                    diveDate={survey.diveDate}
                  />
                </Grid>
              )}
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Paper>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    ...incomingStyles,
    cardImageWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    infoWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    commentsWrapper: {
      maxHeight: "55%",
    },
    comments: {
      height: "100%",
      overflowY: "auto",
    },
    info: {
      height: "100%",
      padding: "0.5rem 0.5rem 0.5rem 1rem",
    },
  });

interface SurveyCardIncomingProps {
  isAdmin: boolean;
  reefId: number;
  survey: SurveyListItem;
}

type SurveyCardProps = SurveyCardIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(SurveyCard);
