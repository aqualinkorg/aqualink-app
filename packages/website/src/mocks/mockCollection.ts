import { CollectionDetails } from "../store/Collection/types";
import { mockReef } from "./mockReef";
import { mockUser } from "./mockUser";

export const mockCollection: CollectionDetails = {
  id: 1,
  isPublic: false,
  name: "Mock Collection",
  user: mockUser,
  reefs: [mockReef],
  reefIds: [1],
};
