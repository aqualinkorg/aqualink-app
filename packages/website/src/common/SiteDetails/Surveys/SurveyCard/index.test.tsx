import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import SurveyCard from ".";
import { mockUser } from "../../../../mocks/mockUser";
import { mockSurvey } from "../../../../mocks/mockSurvey";

const mockStore = configureStore([]);

describe("Survey Card", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      user: {
        userInfo: mockUser,
      },
    });

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <Router>
          <SurveyCard
            pointName="Test Point"
            pointId={1}
            isAdmin
            reefId={0}
            survey={mockSurvey}
          />
        </Router>
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
