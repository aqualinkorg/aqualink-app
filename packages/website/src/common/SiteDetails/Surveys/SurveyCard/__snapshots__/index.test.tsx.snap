// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey Card should render with given state from Redux store 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-surveyCard-5 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
      style="height: 100%;"
    >
      <div
        class="MuiGrid-root makeStyles-cardImageWrapper-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <a
          href="/sites/0/survey_details/46"
        >
          <div
            class="MuiCardMedia-root makeStyles-cardImage-6"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root makeStyles-infoWrapper-10 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <div
          class="MuiGrid-root makeStyles-info-15 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
          >
            <mock-typography
              classname="makeStyles-cardFields-7"
              variant="h6"
            >
              User:
            </mock-typography>
            <mock-typography
              classname="makeStyles-cardValues-8 makeStyles-valuesWithMargin-13"
              variant="h6"
            >
              Joe Doe
            </mock-typography>
          </div>
          <div
            class="MuiGrid-root makeStyles-commentsWrapper-11 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-start MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              style="height: 80%;"
            >
              <mock-typography
                classname="makeStyles-cardFields-7"
                variant="h6"
              >
                Comments:
              </mock-typography>
              <mock-typography
                classname="makeStyles-cardValues-8 makeStyles-comments-12"
                variant="h6"
              >
                No comments
              </mock-typography>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
          >
            <mock-typography
              classname="makeStyles-cardFields-7"
              variant="h6"
            >
              Temp:
            </mock-typography>
            <mock-typography
              classname="makeStyles-cardValues-8 makeStyles-valuesWithMargin-13"
              variant="h6"
            >
              10.0 Â°C
            </mock-typography>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
            >
              <a
                href="/sites/0/survey_details/46"
                style="text-decoration: none;"
              >
                <mock-button
                  color="primary"
                  size="small"
                  variant="outlined"
                >
                  VIEW DETAILS
                </mock-button>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-2"
            >
              <mock-iconbutton>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </mock-iconbutton>
              <mock-dialog
                fullwidth="true"
                open="false"
              >
                <mock-dialogcontent
                  dividers="true"
                >
                  <mock-typography
                    color="textSecondary"
                  >
                    Are you sure you would like to delete the survey for 09/10/2020? It will delete all media associated with this survey.
                  </mock-typography>
                  <div
                    class="MuiCollapse-root DeleteButton-alert-24 MuiCollapse-hidden"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
                          role="alert"
                        >
                          <div
                            class="MuiAlert-icon"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                          </div>
                          <div
                            class="MuiAlert-message"
                          />
                          <div
                            class="MuiAlert-action"
                          >
                            <mock-iconbutton
                              color="inherit"
                              size="small"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                />
                              </svg>
                            </mock-iconbutton>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </mock-dialogcontent>
                <mock-dialogactions>
                  <mock-button
                    color="secondary"
                    size="small"
                    variant="outlined"
                  >
                    Cancel
                  </mock-button>
                  <mock-button
                    color="primary"
                    size="small"
                    variant="outlined"
                  >
                    Yes
                  </mock-button>
                </mock-dialogactions>
              </mock-dialog>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
